### PR TITLE
Update lifecycle version

### PR DESCRIPTION
**Pull request by NotWoods**
The new version [splits the dependency up](https://developer.android.com/jetpack/androidx/releases/lifecycle) into separate libraries to use.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/tab-collections/build.gradle [hunk 1]
```diff
@@ -49,8 +49,7 @@ dependencies {
     implementation Dependencies.kotlin_coroutines
 
     implementation Dependencies.androidx_paging
-    implementation Dependencies.androidx_lifecycle_extensions
-    kapt Dependencies.androidx_lifecycle_compiler
+    implementation Dependencies.androidx_lifecycle_livedata
```
Comment (psymoon) [inline components/feature/tab-collections/build.gradle:52]: 🤔 What do we need `livedata` dependency for?

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -90,6 +90,10 @@ object Dependencies {
     const val androidx_core_ktx = "androidx.core:core-ktx:${Versions.AndroidX.core}"
     const val androidx_fragment = "androidx.fragment:fragment:${Versions.AndroidX.fragment}"
     const val androidx_lifecycle_extensions = "androidx.lifecycle:lifecycle-extensions:${Versions.AndroidX.lifecycle}"
```
Comment (psymoon) [inline buildSrc/src/main/java/Dependencies.kt:92]: Looks like this is deprecated, does this still need to be here?

File: components/support/webextensions/build.gradle [hunk 3]
```diff
@@ -37,7 +37,6 @@ dependencies {
     implementation project(':browser-state')
     implementation project(':support-base')
     implementation project(':support-ktx')
-    implementation Dependencies.androidx_lifecycle_extensions
```
Comment (psymoon) [inline components/support/webextensions/build.gradle:40]: I think this was added due to `WebExtensionPopupFeature`. Can we verify that works? (Sample browser -> install ublock -> click on ublock browser action)

File: components/feature/tab-collections/build.gradle (hunk 1 above)
Comment (NotWoods) [reply components/feature/tab-collections/build.gradle:52]: `androidx_lifecycle_extensions` had both LiveData and ViewModel and was replaced with two separate dependencies. `TabCollectionDao` currently uses `LiveData` to return some results so I left a dependency on `androidx_lifecycle_livedata`.

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (NotWoods) [reply buildSrc/src/main/java/Dependencies.kt:92]: No, I'll remove it

File: components/support/webextensions/build.gradle (hunk 3 above)
Comment (NotWoods) [reply components/support/webextensions/build.gradle:40]: Yes, it still works.

**Comment by psymoon** [comment]
@NotWoods Looks like there are some dependency failures in the CI with `feature-share`:
https://firefoxci.taskcluster-artifacts.net/dP-x-rgMQBKTXo14owasPw/0/public/logs/live_backing.log

Do you also see this locally?

**Comment by NotWoods** [comment]
I don't see the failure locally, but I'll try adding a livedata dependency and see if it helps.

**Comment by NotWoods** [comment]
Actually, it seems like there are a lot of unneeded dependencies. I'll try pulling out some.

**Comment by NotWoods** [comment]
bors r=psymoon

**Pull request merged**